File: frontend/src/types/manpowerQuality.ts
export interface ManpowerSummary {
  required: number;
  available: number;
  shortage: number;
  attendance_pct: number;
}

export interface DepartmentManpowerItem {
  department: string;
  required: number;
  available: number;
  gap: number;
  attendance_pct: number;
}

export interface ManpowerTrendPoint {
  date_label: string;
  required: number;
  available: number;
  shortage: number;
  attendance_pct: number;
}

export interface QualityStatusData {
  status: 'NORMAL' | 'ATTENTION' | 'CRITICAL';
  samples_tested: number;
  pass_rate_pct: number;
  defect_rate_pct: number;
  main_issue: string;
}

export interface QualityParameterItem {
  parameter: string;
  label: string;
  value: number;
  limit: number;
  unit: string;
  status: 'NORMAL' | 'ATTENTION' | 'OUT OF LIMIT';
}

export interface QualityTrendPoint {
  date_label: string;
  value: number;
  limit: number;
}

export interface QualityMachineItem {
  machine_process: string;
  quality_issue: string;
  current_value: number;
  limit: number;
  unit: string;
  status: 'NORMAL' | 'ATTENTION' | 'OUT OF LIMIT';
}

export interface AttentionItem {
  severity: 'HIGH' | 'MEDIUM' | 'LOW';
  title: string;
  detail: string;
}

export interface ManpowerQualityModuleData {
  company_name: string;
  period: string;
  manpower_summary: ManpowerSummary;
  department_gaps: DepartmentManpowerItem[];
  manpower_trend: ManpowerTrendPoint[];
  quality_status: QualityStatusData;
  quality_parameters: QualityParameterItem[];
  quality_trends_by_param: Record<string, QualityTrendPoint[]>;
  quality_issues_by_machine: QualityMachineItem[];
  needs_attention: AttentionItem[];
}

File: backend/app/schemas/manpower_quality.py
from typing import List, Optional
from pydantic import BaseModel

class ManpowerSummary(BaseModel):
    required: int
    available: int
    shortage: int
    attendance_pct: float

class DepartmentManpowerItem(BaseModel):
    department: str
    required: int
    available: int
    gap: int
    attendance_pct: float

class ManpowerTrendPoint(BaseModel):
    date_label: str
    required: int
    available: int
    shortage: int
    attendance_pct: float

class QualityStatusData(BaseModel):
    status: str  # 'NORMAL' | 'ATTENTION' | 'CRITICAL'
    samples_tested: int
    pass_rate_pct: float
    defect_rate_pct: float
    main_issue: str

class QualityParameterItem(BaseModel):
    parameter: str
    label: str
    value: float
    limit: float
    unit: str
    status: str  # 'NORMAL' | 'ATTENTION' | 'OUT OF LIMIT'

class QualityTrendPoint(BaseModel):
    date_label: str
    value: float
    limit: float

class QualityMachineItem(BaseModel):
    machine_process: str
    quality_issue: str
    current_value: float
    limit: float
    unit: str
    status: str  # 'NORMAL' | 'ATTENTION' | 'OUT OF LIMIT'

class AttentionItem(BaseModel):
    severity: str  # 'HIGH' | 'MEDIUM' | 'LOW'
    title: str
    detail: str

class ManpowerQualityModuleResponse(BaseModel):
    company_name: str = "Ashok Textiles"
    period: str = "THIS_MONTH"
    manpower_summary: ManpowerSummary
    department_gaps: List[DepartmentManpowerItem]
    manpower_trend: List[ManpowerTrendPoint]
    quality_status: QualityStatusData
    quality_parameters: List[QualityParameterItem]
    quality_trends_by_param: dict
    quality_issues_by_machine: List[QualityMachineItem]
    needs_attention: List[AttentionItem]
